### DISCUSSION TITLE
How to sync only some but not all folders using sync_list ?

### DISCUSSION THREAD

**Discussion by noobyprogrammer** [abraunegg/onedrive#2446] [Q&A] [answered]
Thank you so much for creating this client! I have many files on my OneDrive and tools like this make migration and maintenance of the cloud storage a lot easier.

However, I still have a few problems as a newbie. They prevent me from using it correctly.

How can I correctly sync only a few folders? 

I reinstalled the client. (Purged everything beforhand, also removed symlinks I found in 
`/home/user/.config/systemd/user/default.target.wants/onedrive.service `
after deinstalling, looked for leftover configuration files to make sure everything was gone).

I reinstalled it because I had systemctl running and my SSD was too small...

I have always done everything the way it was explained on this page for my Linux Mint 21 system: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204

More about my system:
```
$ cat /etc/issue
Linux Mint 21.1 Vera \n \l
$ uname -a
Linux NetacStickB 5.15.0-76-generic #83-Ubuntu SMP Thu Jun 15 19:16:32 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

This time, immediately after my new installation and before doing any "--synchronize" command, I created a sync_list file in `/home/user/.config/onedrive ` and it also showed up in the config. 

```
myname@NetacStickB:/$ onedrive --display-config
onedrive version = v2.4.25-1+np1
Config path = /home/myname/.config/onedrive
Config file found in config path  = false
Config option 'sync_dir'  = /home/myname/OneDrive
Config option 'enable_logging' = false
Config option 'log_dir'  = /var/log/onedrive/
Config option 'disable_notifications'  = false
Config option 'min_notify_changes'  = 5
Config option 'skip_dir'   = 
Config option 'skip_dir_strict_match'  = false
Config option 'skip_file'  = ~*|.~*|*.tmp
Config option 'skip_dotfiles' = false
Config option 'skip_symlinks' = false
Config option 'monitor_interval' = 300
Config option 'monitor_log_frequency'  = 6
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'   = false
Config option 'dry_run' = false
Config option 'upload_only'   = false
Config option 'download_only' = false
Config option 'local_first'   = false
Config option 'check_nosync'  = false
Config option 'check_nomount' = false
Config option 'resync'  = false
Config option 'resync_auth'   = false
Config option 'cleanup_local_files' = false
Config option 'classify_as_big_delete' = 1000
Config option 'disable_upload_validation' = false
Config option 'bypass_data_preservation'  = false
Config option 'no_remote_delete' = false
Config option 'remove_source_files' = false
Config option 'sync_dir_permissions'   = 700
Config option 'sync_file_permissions'  = 600
Config option 'space_reservation'   = 52428800
Config option 'application_id'   = 
Config option 'azure_ad_endpoint'   = 
Config option 'azure_tenant_id'  = common
Config option 'user_agent' = 
Config option 'force_http_11' = false
Config option 'debug_https'   = false
Config option 'rate_limit' = 0
Config option 'operation_timeout'   = 3600
Config option 'dns_timeout'   = 60
Config option 'connect_timeout'  = 10
Config option 'data_timeout'  = 600
Config option 'ip_protocol_version' = 0
Config option 'sync_root_files'  = false
Selective sync 'sync_list' configured  = true
sync_list contents:
/Documents/subfolder/subfolder_to_sync
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured  = false
Config option 'webhook_enabled'  = false
```

If I try to run `onedrive --synchronize --resync` as suggested in the usage doc, it wants to sync everything, not just the specified folders.

```
myname@NetacStickB:/$ sudo onedrive --synchronize --resync --dry-run
[sudo] password for myname:                 
The use of --resync will remove your local 'onedrive' client state, thus no record will exist regarding your current 'sync status'
This has the potential to overwrite local versions of files with potentially older versions downloaded from OneDrive which can lead to data loss
If in-doubt, backup your local data first before proceeding with --resync
Are you sure you wish to proceed with --resync? [Y/N] y
DRY-RUN Configured. Output below shows what 'would' have occurred.
Deleting the saved application sync status ...
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Syncing changes and items from OneDrive ...
Downloading file ./presentation.pptx ... done.
Creating local directory: ./Documents
Creating local directory: Documents/afolder
Creating local directory: Documents/Subfolder
Creating local directory: Documents/Subfolder/anotherSubfolder
Creating local directory: Documents/Subfolder/anotherSubfolder/yetAnotherFolder23
Creating local directory: Documents/Subfolder/anotherSubfolder/yetAnotherFolder23/i_dont_want_this_on_my_SSD
Creating local directory: Documents/Subfolder/anotherSubfolder/yetAnotherFolder23/another_folder_I_dont_want_on_my_SSD
etc.
etc.
etc.
```

I have also tried putting --resync in front of --synchronized.

I also had the problem of the systemd service randomly starting (inspite of using `--user disable onedrive.service`) - I verified via `systemctl --user status onedrive.service`  to make sure it was not the monitor running - although I believe this is a systemd process as well?
However, unlike with the --synchronize command, which wanted to synchronize the whole OneDrive, this time the specified folder was synced. But I don't seem to get it to really work.

I thought maybe the problem was "Config file found in config path  = false" in the config, but, then again, it seems to know the folders I specified in the sync_list when the service is run in the background. The manual calls are the things that I don't get to work.

I only want to sync manually with the --synchronize command and maybe via the --monitor command.

Maybe someone on here can explain to me how to use it correctly, because I've spent days trying to configure it, but I'm too much of a Linux novice, unfortunately.

**Comment by abraunegg** [comment, marked as answer]
@noobyprogrammer 

Please can you re-read the documentation more carefully, as this is your issue:
```
myname@NetacStickB:/$ sudo onedrive --synchronize --resync --dry-run
```

You are using `sudo` which you should not be using - as this elevates your user to 'root' - running the application as 'root' not the user you want to run the application as. Please stop adding `sudo` to your command line as it is not required.

**Reply by noobyprogrammer** [reply]
Thank you so much for your answer. What you explained makes sense to me.

 I reinstalled everything, have not started any systemd services yet. I created a dummy directory `/Documents/subfolder/subfolder_two/ ` and ran `onedrive --synchronize --resync --dry-run` (although I don't know if the `--resync ` flag was necessary since I hadn't synchronized it in the first place due to my OneDrive having more than 500 GB of data). 

I was getting some API warnings, though:
```
$ onedrive --synchronize --resync --dry-run

The use of --resync will remove your local 'onedrive' client state, thus no record will exist regarding your current 'sync status'
This has the potential to overwrite local versions of files with potentially older versions downloaded from OneDrive which can lead to data loss
If in-doubt, backup your local data first before proceeding with --resync

Are you sure you wish to proceed with --resync? [Y/N] y

DRY-RUN Configured. Output below shows what 'would' have occurred.
Deleting the saved application sync status ...
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Syncing changes and items from OneDrive ...
Creating local directory: ./Documents
Creating local directory: Documents/subfolder
WARNING: OneDrive API inconsistency - this file does not have any hash: Documents/subfolder/Dokument1.docx
Creating local directory: Documents/subfolder/subfolder_two
Creating local directory: Documents/subfolder/subfolder_two/subfolder_two_2
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129239
WARNING: OneDrive API inconsistency - this file does not have any hash: Documents/subfolder/subfolder_two/Dokument1.docx
Downloading file Documents/subfolder/subfolder_two/Dokument1.docx ... done.
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129238
Creating local directory: Documents/subfolder/subfolder_two/subfolder_two_1



etc. 
etc.
```

I then ran  `onedrive --synchronize --resync --dry-run` with  one of the "real" directories I want to sync in sync_list instead.
However, a lot of warnings along the lines of the following were displayed again:
```
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129235
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129239
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129245
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129238
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129247
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129248
WARNING: OneDrive API inconsistency - this file does not have any hash: EB356FD8AA88F1C3!129251
Creating local directory: 

etc. etc. etc.


```
After some time passed, the files started to download.

Are the API inconsistency warnings due to recent changes I made to some files and folders, like renaming? Still, "OneDrive API inconsistency - this file does not have any hash" pertained to files I hadn't touched in more than 4 years.

**Comment by abraunegg** [comment]
@noobyprogrammer 
Regarding the warnings, please read in detail:
* https://github.com/abraunegg/onedrive/issues/2433
* https://github.com/abraunegg/onedrive/discussions/2441

**Reply by noobyprogrammer** [reply]
Thanks! I don't have a business account, but the second link cleared it up. I am glad already have the latest 2.4.25 version.
It appeared for all kinds of files, I don't have any OneNote items.
